model/api_operations.js

```diff
@@ -107,9 +108,9 @@ function create_channel(user_id)
         }else {
             // if he is host
             // destroy all the video in the channel, delete channel, create a new channel
-            return model.video.destroy({ where: {channel: result.id}})
-            .then(() => {model.channel.destroy({ where: {host: user_id}})})
-            .then(() => {model.channel.create({ host : user_id })})
+            return model.video.destroy({ where: { channel: result.id}})
+            .then(() => {model.channel.destroy({ where: { host: result.host}})})
+            .then(() => {model.channel.create({ host: result.host})})
```

Comment:
The promise is not being returned, so nothing will be passed to next `then`. Add `return` before the `model.channel...` or remove the curly braces around them.